### PR TITLE
Mitochondrial SNV pipeline PR 2 of N

### PR DESCRIPTION
**Pull request by cassimons**
Stacked on #309.

This PR builds on #309, adding a stage that generates the final per-sequence group variant calls.

The steps in this stage include:
- Using mutect to call variants in both the wt chrM AND the shifted chrM
- Merging the two variant calls to generate a single vcf covering the full chrM
- Merging the mutect output stats files 
- Initial round of filtering to exclude blacklist and high alt alleles variants
- Running the haplocheck tool to get one estimate of contamination
- Evaluation of contamination estimates from both haplocheck and verifybamid (run in the main pipeline)
- Run a final round of filtering to exclude variants with a VAF below the estimated contamination level.

### CODE REVIEW

File: configs/defaults/seqr_loader.toml [hunk 1]
```diff
@@ -91,3 +91,6 @@ f_score_beta = 1.0
 # Sarah Stenton from Broad runs this pipline for seqr ingest and indicated they use a
 # threshold of 0.01 for seqr cohorts.
 vaf_filter_threshold = 0.01
+# Use verifybamid in adition to haplocheck for contamination estimate
```
Comment (vivbak) [inline configs/defaults/seqr_loader.toml:94]:
```suggestion
# Use verifybamid in addition to haplocheck for contamination estimate
```

File: cpg_workflows/defaults.toml [hunk 2]
```diff
@@ -92,3 +92,14 @@ SYNDIP = 'syndip'
...
+
+[mito_snv]
+# Example config for broad wdl found here:
+# https://raw.githubusercontent.com/broadinstitute/gatk/master/scripts/mitochondria_m2_wdl/ExampleInputsMitochondriaPipeline.json
+# f_score_beta is not configured so will use tool default of 1.0
+f_score_beta = 1.0
+# Sarah Stenton from Broad runs this pipline for seqr ingest and indicated they use a
+# threshold of 0.01 for seqr cohorts.
+vaf_filter_threshold = 0.01
+# Use verifybamid in adition to haplocheck for contamination estimate
```
Comment (vivbak) [inline cpg_workflows/defaults.toml:104]:
```suggestion
# Use verifybamid in addition to haplocheck for contamination estimate
```

File: cpg_workflows/stages/mito.py [hunk 3]
```diff
@@ -259,3 +262,221 @@ def queue_jobs(
...
+            shifted_vcf=shifted_call_j.output_vcf,
+            reference=mito_ref,
+            shift_back_chain=shifted_mito_ref.shift_back_chain,
+            job_attrs=self.get_job_attrs(sequencing_group),
+        )
+        jobs.append(merge_j)
+        assert isinstance(merge_j.output_vcf, hb.ResourceGroup)
+
+        # Merge the mutect stats output files (needed for filtering)
+        merge_stats_J = mito.merge_mutect_stats(
```
Comment (vivbak) [inline cpg_workflows/stages/mito.py:385]: nit? not sure if intentional 
```suggestion
        merge_stats_j = mito.merge_mutect_stats(
```

File: cpg_workflows/stages/mito.py [hunk 4]
```diff
@@ -259,3 +262,221 @@ def queue_jobs(
...
+
+        # Initial round of filtering to exclude blacklist and high alt alleles
+        initial_filter_j = mito.filter_variants(
+            b=get_batch(),
+            vcf=merge_j.output_vcf,
+            reference=mito_ref,
+            merged_mutect_stats=merge_stats_J.combined_stats,
+            # alt_allele and vaf config hardcoded in this round of filtering as per
+            # https://github.com/broadinstitute/gatk/blob/master/scripts/mitochondria_m2_wdl/AlignAndCall.wdl#L167
+            max_alt_allele_count=4,
```
Comment (vivbak) [inline cpg_workflows/stages/mito.py:402]: Would we be able to define these hardcoded variables at the top of the page?

File: cpg_workflows/stages/mito.py [hunk 5]
```diff
@@ -259,3 +262,221 @@ def queue_jobs(
...
+        jobs.append(parse_contamination_j)
+
+        # Filter round 2 - remove variants with VAF below estimated contamination
+        second_filter_j = mito.filter_variants(
+            b=get_batch(),
+            vcf=initial_filter_j.output_vcf,
+            reference=mito_ref,
+            merged_mutect_stats=merge_stats_J.combined_stats,
+            # alt_allele config from https://github.com/broadinstitute/gatk/blob/master/scripts/mitochondria_m2_wdl/AlignAndCall.wdl#L167
+            max_alt_allele_count=4,
```
Comment (vivbak) [inline cpg_workflows/stages/mito.py:450]: As above

File: cpg_workflows/stages/mito.py (hunk 3 above)
Comment (cassimons) [reply cpg_workflows/stages/mito.py:385]: Nope, just poor consistency on my part 😅 I will change them.

File: cpg_workflows/stages/mito.py (hunk 4 above)
Comment (cassimons) [reply cpg_workflows/stages/mito.py]: done.

**Review by vivbak — APPROVED** [review]
Joint review by me & @EddieLF

**Pull request merged**